src/app/clients/ActionBar.tsx

```diff
@@ -143,28 +174,48 @@ const ActionBar: React.FC<Props> = ({ selected, data }) => {
                     })}
                 </MenuList>
             </Menu>
-            <Menu
-                open={actionAnchorElement !== null}
-                onClose={() => setActionAnchorElement(null)}
-                anchorEl={actionAnchorElement}
-            >
-                <MenuList id="action-menu">
-                    {actions.map((action, index) => {
-                        return (
-                            <MenuItem
-                                key={index}
-                                onClick={() => {
-                                    setSelectedAction(action);
-                                    setActionModal(true);
+            {actionAnchorElement ? (
+                <Menu
+                    open={actionAnchorElement !== null}
+                    onClose={() => setActionAnchorElement(null)}
+                    anchorEl={actionAnchorElement}
+                >
+                    <MenuList id="action-menu">
+                        <MenuItem
+                            onClick={() => {
+                                if (selectedData.length === 0) {
                                     setActionAnchorElement(null);
-                                }}
-                            >
-                                {action}
-                            </MenuItem>
-                        );
-                    })}
-                </MenuList>
-            </Menu>
+                                    setModalError("Please select rows for printing.");
+                                } else {
+                                    setSelectedAction("Print Shipping Labels");
+                                    setShippingLabelsModal(true);
+                                    setActionAnchorElement(null);
+                                    setModalError(null);
+                                }
+                            }}
+                        >
+                            Print Shipping Labels
+                        </MenuItem>
+                        <MenuItem
```

Comment:
Can these all be combined into one element with different props? Maybe a later job but I don't see why you need the different sections.

Reply:
Yes I will refactor this in a future ticket.